iml-services/iml-journal/src/lib.rs

```diff
@@ -0,0 +1,382 @@
+    pool: &'a PgPool,
+    msg: &str,
+    _: i32,
+    _: i32,
+) -> HandlerFut<'a> {
+    let (lustre_pid, msg) = match server_security_flavor_parser(msg) {
+        Some(x) => x,
+        None => return future::ok(()).boxed(),
+    };
+
+    struct Row {
+        id: i32,
+        message: Option<String>,
+    }
+
+    async move {
+        let row = sqlx::query_as!(Row,
+            "SELECT id, message FROM chroma_core_alertstate WHERE lustre_pid = $1 ORDER BY id DESC LIMIT 1",
+            Some(lustre_pid)
+        )
+        .fetch_optional(pool)
+        .await?;
+
+        let (id, msg) = match row {
+            Some(Row { id, message: Some(message) }) => (id, format!("{} {}", message, msg)),
+            Some(Row { message:None, ..}) | None => return Ok(()),
+        };
+
+        sqlx::query!(
+            r#"
+            UPDATE chroma_core_alertstate
+            SET message = $1
+            WHERE
+                id = $2
+        "#,
+            msg,
+            id
+        )
+        .execute(pool)
+        .await?;
+
+        Ok(())
+    }
+    .boxed()
+}
+
+fn admin_client_eviction_parser(msg: &str) -> Option<(i32, String)> {
+    let uuid = get_item_after(msg, "evicting ")?;
+    let x = format!("client {} evicted by the administrator", uuid);
+    let lustre_pid = msg[9..9 + msg[9..].find(":")?].parse::<i32>().ok()?;
+
+    Some((lustre_pid, x))
+}
+
+fn admin_client_eviction_handler<'a>(
+    pool: &'a PgPool,
+    msg: &str,
+    host_id: i32,
+    host_content_type_id: i32,
+) -> HandlerFut<'a> {
+    if let Some((lustre_pid, msg)) = admin_client_eviction_parser(msg) {
+        return alert::raise(
+            pool,
+            AlertRecordType::ClientConnectEvent,
+            msg,
+            host_content_type_id,
+            Some(lustre_pid),
+            AlertSeverity::WARNING,
+            host_id,
+        )
+        .err_into()
+        .boxed();
+    };
+
+    future::ok(()).boxed()
+}
+
+fn client_eviction_parser(msg: &str) -> Option<(i32, String)> {
+    let s = msg.find("### ")? + 4;
+    let l = msg[s..].find(": evicting client at ")?;
+    let reason = &msg[s..s + l];
+    let client = get_item_after(msg, ": evicting client at ")?;
+    let lustre_pid = get_item_after(msg, "pid: ")?.parse::<i32>().ok()?;
+
+    Some((lustre_pid, format!("client {} evicted: {}", client, reason)))
+}
+
+fn client_eviction_handler<'a>(
+    pool: &'a PgPool,
+    msg: &str,
+    host_id: i32,
+    host_content_type_id: i32,
+) -> HandlerFut<'a> {
+    if let Some((lustre_pid, msg)) = client_eviction_parser(msg) {
+        return alert::raise(
+            pool,
+            AlertRecordType::ClientConnectEvent,
+            msg,
+            host_content_type_id,
+            Some(lustre_pid),
+            AlertSeverity::WARNING,
+            host_id,
+        )
+        .err_into()
+        .boxed();
+    };
+
+    future::ok(()).boxed()
+}
+
+fn get_item_after<'a>(s: &'a str, after: &str) -> Option<&'a str> {
```

Comment:
I think [strip_prefix](https://doc.rust-lang.org/beta/std/primitive.str.html#method.strip_prefix) is close to this.

Reply:
Not worth fixing. Once there are more matches, plan is to switch to parser-combinator a-la https://docs.rs/combine/4.2.1/combine/.